packages/adapter/src/response-processor/caster.ts

```diff
@@ -75,13 +73,21 @@ export function castValue(value: unknown, type: ResponseType) {
   }
 }
 
-export function multiplyValue(value: number | string, times: number | string): string {
+export function multiplyValue(value: string | BigNumber, times: number | string): string {
```

Comment:
The `times` here is `encoded` in the request `parameters` and may or may not be a big number. For example, it is easy to think of a use case where it would be `10^18`. But I guess we depend on `cbor.js` to decode it, which can't do big numbers? Tomorrow I will try encoding a 256 bit number and decoding it with cbor-custom to see what it spits out.

Reply:
I decoded `{key: 1000000000000000000}` using `cbor-custom` and received a big number.  So we can also avoid using `number` for `_times` here. lmk what you think.

Reply:
Good point. I don't think we should be using regular JS numbers at all for anything number related. Either strings or big numbers